src/modules/configurator/hooks/useConfigProperty.tsx

```diff
@@ -0,0 +1,74 @@
+import { useEffect, useState } from "react";
+import { fetchPropertyImagesByVariant } from "../../../services/API_configurations";
+import {
+  dropdownState,
+  dropdownOpen,
+  userConfigurationState,
+  activePropState,
+} from "../state";
+import { useSetRecoilState, useRecoilState } from "recoil";
+
+interface PropertyProps {
+  index: number;
+  propertyName: "color" | "wheels";
+  modelName: "Audi RS5" | "Audi RS6" | "Audi e-tron GT" | undefined;
+  label: string | number;
+  name: string;
+  description: string;
+  price: number;
+}
+
+export function useConfigProperty({
+  index,
+  propertyName,
+  modelName,
+  label,
+  name,
+  description,
+  price,
+}: PropertyProps) {
+  const [propertyImgUrl, setPropertyImgUrl] = useState("");
+  const setActiveDropdownName = useSetRecoilState(dropdownState);
+  const [isDropdownOpen, setIsDropdownOpen] = useRecoilState(dropdownOpen);
+  const [currentUserConfiguration, setCurrentUserConfiguration] =
+    useRecoilState(userConfigurationState);
+
+  const [activePropIndex, setActivePropIndex] = useRecoilState(activePropState);
+
+  useEffect(() => {
+    handleImageFetch();
+  }, []);
```

Comment:
What if the model name changes or you dont get the model name on mount and the functions doesnt trigger?
Add model name inside the dependency array.

Reply:
Added model name to a dependency array.